recipe/meta.yaml

```diff
@@ -1,60 +1,116 @@
-    - test -f ${PREFIX}/include/re2/re2.h  # [unix]
-    - if not exist %PREFIX%\\Library\\include\\re2\\re2.h exit 1  # [win]
-    - if not exist %PREFIX%\\Library\\lib\\re2.lib exit 1  # [win]
-    - if exist %PREFIX%\\Library\\lib\\re2_static.lib exit 1  # [win]
-    - if not exist %PREFIX%\\Library\\bin\\re2.dll exit 1  # [win]
+    # only for build testing; these don't have run-exports
+    - benchmark  # [build_platform == target_platform]
+    - gtest      # [build_platform == target_platform]
+
+outputs:
+  # re2 has a rarely-changing SOVERSION, and it's enough to depend on
+  # (and run-export) the output that contains the library artefact
+  - name: re2
+    script: install.sh   # [unix]
+    script: install.bat  # [win]
+    build:
+      run_exports:
+        - {{ pin_subpackage("libre2-" ~ soversion) }}
+        # by adding this run-export, we avoid that different libre2-{X,Y} builds
+        # can be co-installed, because they have a non-overlapping range for the
+        # allowed re2-version (see below) and each re2 requires a specific soversion
+        - re2
+    requirements:
+      build:
+        - cmake
+      host:
+        - {{ pin_subpackage("libre2-" ~ soversion, exact=True) }}
+      run:
+        - {{ pin_subpackage("libre2-" ~ soversion, exact=True) }}
+    test:
+      requires:
+        - {{ compiler('cxx') }}
+        - cmake
+        - ninja
+        - pkg-config
+      files:
+        - cmake_test/
+      commands:
+        # presence of shared libs (unix) & import lib (win)
+        - test -f ${PREFIX}/lib/libre2.so               # [linux]
+        - test -f ${PREFIX}/lib/libre2.dylib            # [osx]
+        - if not exist %LIBRARY_LIB%\re2.lib exit 1     # [win]
+
+        # absence of static libs
+        - test ! -f ${PREFIX}/lib/libre2.a              # [unix]
+        - if exist %LIBRARY_LIB%\re2_static.lib exit 1  # [win]
+
+        # presence of header
+        - test -f ${PREFIX}/include/re2/re2.h           # [unix]
+        - if not exist %LIBRARY_INC%\re2\re2.h exit 1   # [win]
+
+        # test correctness of pkg-config metadata
+        - pkg-config --print-errors --exact-version "{{ soversion }}.0.0" re2
+
+        # CMake integration
+        - cd cmake_test
+        - cmake -GNinja -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_STANDARD=17 .
+        - cmake --build .
+        - ./re2_example    # [unix]
+        - re2_example.exe  # [win]
+
+    # the versioned library artefact
+  - name: libre2-{{ soversion }}
+    script: install.sh   # [unix]
+    script: install.bat  # [win]
+    build:
+      run_exports:
+        - {{ pin_subpackage("libre2-" ~ soversion) }}
+    requirements:
+      build:
+        # for strong run-exports
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+      host:
+        - libabseil
+      run_constrained:
+        # keep expanding the valid range until the soversion gets bumped
+        - re2 >={{ soversion_last_updated }},<={{ dotversion }}
```

Comment:
This should be,
```
re2 {{ dotversion }}.*
```

Reply:
It works out to be the same (since `re2` depends exactly on `libre2-X`, it can only ever be satisfied by the upper bound), but would have saved some indirection. However, since this lets us get rid of `soversion_last_updated`, that's a net benefit.